src/pyuvdata/beam_interface.py

```diff
@@ -82,6 +90,121 @@ def __post_init__(self, include_cross_pols: bool):
                     "efield beam, either provide an efield UVBeam or do not "
                     "specify `beam_type`."
                 )
+        elif self.beam_type is None:
+            self.beam_type = "efield"
+
+    @property
+    def Npols(self):  # noqa N802
+        """The number of polarizations in the beam."""
+        return self.beam.Npols or len(self.polarization_array)
+
+    @property
+    def polarization_array(self):
+        """The polarizations defined on the beam."""
+        return self.beam.polarization_array
+
+    @property
+    def feed_array(self):
+        """The feeds for which the beam is defined."""
+        return self.beam.feed_array
+
+    @property
+    def Nfeeds(self):  # noqa N802
+        """The number of feeds defined on the beam."""
+        return self.beam.Nfeeds or len(self.feed_array)
+
+    def clone(self, **kw):
+        """Return a new instance with updated parameters."""
+        return replace(self, **kw)
+
+    def as_power_beam(
+        self, include_cross_pols: bool | None = None, allow_beam_mutation: bool = False
+    ):
+        """Return a new interface instance that is in the power-beam mode.
+
+        If already in the power-beam mode, this is a no-op. Note that this might be
+        slighty unexpected, because the effect of `include_cross_pols` is not accounted
+        for in this case.
+
+        Parameters
+        ----------
+        include_cross_pols : bool, optional
+            Whether to include cross-pols in the power beam. By default, this is True
+            for E-field beams, and takes the same value as the existing beam if the
+            existing beam is already a power beam.
+        allow_beam_mutation : bool, optional
+            Whether to allow the underlying beam to be updated in-place.
+        """
+        if self.beam_type == "power":
+            if include_cross_pols is None:
+                # By default, keep the value of include_cross_pols the same.
+                include_cross_pols = self.Npols > 2
+
+            if self.Npols > 1 and (
+                (include_cross_pols and self.Npols != 4)
+                or (not include_cross_pols and self.Npols != 2)
+            ):
+                warnings.warn(
+                    "as_power_beam does not modify cross pols when the beam is"
+                    "already in power mode!"
```

Comment:
I like this. I feel like it might be more helpful to tell the user what the cross_pol status of their existing object is vs what they asked for.

Reply:
Great idea, done!